RadarSDK/RadarAPIClient.m

```diff
@@ -932,6 +932,51 @@ - (void)getMatrixFromOrigins:(NSArray<CLLocation *> *)origins
                     }];
 }
 
+- (void)sendEvent:(NSString *)name
+     withMetadata:(NSDictionary *_Nullable)metadata
+             user:(RadarUser *_Nullable)user
+completionHandler:(RadarSendEventsAPICompletionHandler _Nonnull)completionHandler {
+    NSString *publishableKey = [RadarSettings publishableKey];
+    if (!publishableKey) {
+        return completionHandler(RadarStatusErrorPublishableKey, nil, nil);
+    }
+
+    NSMutableDictionary *params = [NSMutableDictionary new];
+    params[@"id"] = [RadarSettings _id];
+    params[@"installId"] = [RadarSettings installId];
+    params[@"userId"] = [RadarSettings userId];
+    params[@"deviceId"] = [RadarUtils deviceId];
+
+    params[@"type"] = name;
+    params[@"metadata"] = metadata;
+
+    NSString *host = [RadarSettings host];
+    NSString *url = [NSString stringWithFormat:@"%@/v1/events", host];
+    url = [url stringByAddingPercentEncodingWithAllowedCharacters:[NSCharacterSet URLQueryAllowedCharacterSet]];
+
+    NSDictionary *headers = [RadarAPIClient headersWithPublishableKey:publishableKey];
+
+    [self.apiHelper requestWithMethod:@"POST"
+                                  url:url
+                              headers:headers
+                               params:nil
```

Comment:
Should this be `params: params` instead of `params:nil` since this is a post request?

Reply:
Yep! This was the source of the bad request error. I changed this to `params` and got a success:
```
Send event: status = SUCCESS; location = Optional(<+37.78583400,-122.40641700> +/- 5.00m (speed -1.00 mps / course -1.00) @ 7/19/22, 11:24:01 AM Eastern Daylight Time); events = Optional([<RadarEvent: 0x600001e05100>]); user = Optional(<RadarUser: 0x60000077c420>)
```